poetry/version/markers.py

```diff
@@ -628,7 +628,15 @@ def __hash__(self):
         return h
 
     def __str__(self):
-        return " or ".join(str(m) for m in self._markers)
+        return " or ".join(
+            str(m) for m in self._markers if not m.is_any() and not m.is_empty()
+        )
+
+    def is_any(self):
+        return all(m.is_any() for m in self._markers)
```

Comment:
Since it's a union only one of the markers needs to return `True` for `is_any()`.
```suggestion
        return any(m.is_any() for m in self._markers)
```